android/app/src/main/java/com/microsoft/reactnativecodepush/CodePush.java

```diff
@@ -0,0 +1,412 @@
+    private final String CODE_PUSH_PREFERENCES = "CodePush";
+    private final String DOWNLOAD_PROGRESS_EVENT_NAME = "CodePushDownloadProgress";
+
+    private CodePushPackage codePushPackage;
+    private CodePushReactPackage codePushReactPackage;
+    private CodePushNativeModule codePushNativeModule;
+    private CodePushConfig codePushConfig;
+
+    private Activity mainActivity;
+    private Context applicationContext;
+
+    public CodePush(String deploymentKey, Activity mainActivity) {
+        SoLoader.init(mainActivity, false);
+        this.deploymentKey = deploymentKey;
+        this.codePushPackage = new CodePushPackage(mainActivity.getFilesDir().getAbsolutePath());
+        this.mainActivity = mainActivity;
+        this.applicationContext = mainActivity.getApplicationContext();
+        this.codePushConfig = new CodePushConfig(deploymentKey, this.applicationContext);
+        checkForPendingUpdate(/*needsRestart*/ false);
+    }
+
+    public ReactPackage getReactPackage() {
+        if (codePushReactPackage == null) {
+            codePushReactPackage = new CodePushReactPackage();
+        }
+        return codePushReactPackage;
+    }
+
+    public String getDocumentsDirectory() {
+        return codePushPackage.getDocumentsDirectory();
+    }
+
+    public String getBundleUrl(String assetsBundleFileName) {
+        this.assetsBundleFileName = assetsBundleFileName;
+        String binaryJsBundleUrl = ASSETS_BUNDLE_PREFIX + assetsBundleFileName;
+        try {
+            String packageFile = codePushPackage.getCurrentPackageBundlePath();
+            if (packageFile == null) {
+                // There has not been any downloaded updates.
+                return binaryJsBundleUrl;
+            }
+
+            return packageFile;
+        } catch (IOException e) {
+            throw new CodePushUnknownException("Error in getting current package bundle path", e);
+        }
+    }
+
+    private void cancelRollbackTimer() {
+        if(timer != null) {
+            timer.cancel();
+            timer = null;
+        }
+    }
+
+    private void checkForPendingUpdate(boolean needsRestart) {
+        SharedPreferences settings = applicationContext.getSharedPreferences(CODE_PUSH_PREFERENCES, 0);
+        String pendingUpdateString = settings.getString(PENDING_UPDATE_KEY, null);
+
+        if (pendingUpdateString != null) {
+            try {
+                JSONObject pendingUpdateJSON = new JSONObject(pendingUpdateString);
+                String pendingHash = pendingUpdateJSON.getString(PENDING_UPDATE_HASH_KEY);
+                String currentHash = codePushPackage.getCurrentPackageHash();
+                if (pendingHash.equals(currentHash)) {
+                    int rollbackTimeout = pendingUpdateJSON.getInt(PENDING_UPDATE_ROLLBACK_TIMEOUT_KEY);
+                    initializeUpdateWithRollbackTimeout(rollbackTimeout, needsRestart);
+
+                    settings.edit().remove(PENDING_UPDATE_KEY).commit();
```

Comment:
Will throw exception if it is different instead. http://stackoverflow.com/questions/23113497/what-are-buildconfig-debug-conditional-checks